### PR TITLE
Enable `IndexPath` and `SelectionModel` runtime tests

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): part of #8261

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Tests

## What is the current behavior?

`IndexPath` and `SelectionModel` tests are disabled.

## What is the new behavior?

- Tests are enabled
- Fixes for invalid `SelectedItems` enumeration

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/CustomProperty.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+	public CustomProperty(string name, Type type, Func<object, object> getter, Action<object, object> setter)
+	{
+		Name = name;
+		Type = type;
+		_getter = getter;
+		_setter = setter;
+	}
+
+	public bool CanRead => _getter != null;
```
Comment (ajpinedam) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/CustomProperty.cs:25]: I see `_getter` does not change its value after the object is instantiated. Would not be better to assign the value at the constructor?
```suggestion
	public bool CanRead { get; }
```

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/CustomProperty.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+	public bool CanWrite => _setter != null;
```
Comment (ajpinedam) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/CustomProperty.cs:27]: same as `CanRead`

```suggestion
	public bool CanWrite { get; }
```

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/CustomProperty.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+{
+	private Func<object, object>? _getter;
+	private Action<object, object>? _setter;
+
+	public CustomProperty(string name, Type type, Func<object, object> getter, Action<object, object> setter)
+	{
+		Name = name;
+		Type = type;
+		_getter = getter;
+		_setter = setter;
```
Comment (ajpinedam) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/CustomProperty.cs:22]:
```suggestion
		_setter = setter;
		CanRead = _getter != null;
		CanWrite = _setter != null;
```

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/CustomProperty.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+using System;
+using Windows.UI.Xaml.Data;
+
+namespace Microsoft.UI.Xaml.Controls;
+
+internal class CustomProperty : ICustomProperty
+{
+	private Func<object, object>? _getter;
+	private Action<object, object>? _setter;
```
Comment (ajpinedam) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/CustomProperty.cs:15]:
```suggestion
	private readonly Action<object, object>? _setter;
```

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/CustomProperty.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+#nullable enable
+
+using System;
+using Windows.UI.Xaml.Data;
+
+namespace Microsoft.UI.Xaml.Controls;
+
+internal class CustomProperty : ICustomProperty
+{
+	private Func<object, object>? _getter;
```
Comment (ajpinedam) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/CustomProperty.cs:14]:
```suggestion
	private readonly Func<object, object>? _getter;
```

**Review by ajpinedam — APPROVED** [review]

**Pull request merged**